components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/controller/ODataController.java

```diff
@@ -87,12 +88,16 @@ private JPAODataClaimsProvider createClaims(final Token token, String tenantID)
         }
 
         final JPAClaimsPair<UUID> tenantIDJPAPair;
+        final JPAClaimsPair<UUID> destinationTenantJPAPair;
         if (shouldUseDefaultTenant) {
             tenantIDJPAPair = new JPAClaimsPair<>(UUID.fromString(DEFAULT_TENANT_ID));
+            destinationTenantJPAPair = new JPAClaimsPair<>(UUID.fromString(DEFAULT_TENANT_ID));
```

Comment:
The logic for setting the `destination_tenant_id claim` should not be integrated with the `shouldUseDefaultTenant` logic. 

Currently, if we are in the consumer-provider flow we will use the default tenant (because we want formation filtering) but for the `destination_tenant_id` we will use the actual consumer tenant (because we want to be sure that the destinations will be returned only for the right caller because they are sensitive).

With that change here, you will affect the current consumer-provider flow by saying that whenever we are in it, we will set the `destination_tenant_id` to the default tenant which is incorrect.

We should extract the setting of the `destination_tenant_id` as it was prior this PR (before the logic of the `shouldUseDefaultTenant`):

- check if `tenantID == ""` -> if yes, set the claim to be the `UUID.fromString(DEFAULT_TENANT_ID)` -> else, set it to `UUID.fromString(tenantID)`
  - we still have to check if the tenantID is empty even after the follow up PR because it is a valid use-case for an application not to have `global_subaccount_id` label which will result in empty tenant returned (fully non BTP flow)